nixos/modules/security/acme/default.nix

```diff
@@ -180,607 +218,719 @@ let
+          '';
+        };
 
-      credentialFiles = mkOption {
-        type = types.attrsOf (types.path);
-        inherit (defaultAndText "credentialFiles" {}) default defaultText;
-        description = ''
-          Environment variables suffixed by "_FILE" to set for the cert's service
-          for your selected dnsProvider.
-          To find out what values you need to set, consult the documentation at
-          <https://go-acme.github.io/lego/dns/> for the corresponding dnsProvider.
-          This allows to securely pass credential files to lego by leveraging systemd
-          credentials.
-        '';
-        example = literalExpression ''
-          {
-            "RFC2136_TSIG_SECRET_FILE" = "/run/secrets/tsig-secret-example.org";
-          }
-        '';
-      };
+        postRun = mkOption {
+          type = types.lines;
+          inherit (defaultAndText "postRun" "") default defaultText;
+          example = "cp full.pem backup.pem";
+          description = ''
+            Commands to run after new certificates go live. Note that
+            these commands run as the root user.
 
-      dnsPropagationCheck = mkOption {
-        type = types.bool;
-        inherit (defaultAndText "dnsPropagationCheck" true) default defaultText;
-        description = ''
-          Toggles lego DNS propagation check, which is used alongside DNS-01
-          challenge to ensure the DNS entries required are available.
-        '';
-      };
+            Executed in the same directory with the new certificate.
+          '';
+        };
+
+        keyType = mkOption {
+          type = types.str;
+          inherit (defaultAndText "keyType" "ec256") default defaultText;
+          description = ''
+            Key type to use for private keys.
+            For an up to date list of supported values check the --key-type option
+            at <https://go-acme.github.io/lego/usage/cli/options/>.
+          '';
+        };
+
+        dnsProvider = mkOption {
+          type = types.nullOr types.str;
+          inherit (defaultAndText "dnsProvider" null) default defaultText;
+          example = "route53";
+          description = ''
+            DNS Challenge provider. For a list of supported providers, see the "code"
+            field of the DNS providers listed at <https://go-acme.github.io/lego/dns/>.
+          '';
+        };
+
+        dnsResolver = mkOption {
+          type = types.nullOr types.str;
+          inherit (defaultAndText "dnsResolver" null) default defaultText;
+          example = "1.1.1.1:53";
+          description = ''
+            Set the resolver to use for performing recursive DNS queries. Supported:
+            host:port. The default is to use the system resolvers, or Google's DNS
+            resolvers if the system's cannot be determined.
+          '';
+        };
+
+        tlsMode = lib.mkOption {
+          type = lib.types.bool;
+          inherit (defaultAndText "tlsMode" false) default defaultText;
```

Comment:
@Atemu Turns out `mkEnableOption` couldn't have been used here anyway, because of that defaults-system.

Reply:
It's possible to `mkEnableOption ... // defaultsAndText ...`.

Reply:
Okay but there are many more occurrences of the same pattern and I don't think it makes sense to clobber this PR (this is the wrong one btw for me internally) with such refactoring.